Helper/Cart.php

```diff
@@ -497,58 +497,67 @@ public function getCartData($payment_only, $place_order_payload, $quote = null)
-            $shippingAddress->save();
-
-            // Shipping address
-            $shipping_address = [
-                'first_name'      => $shippingAddress->getFirstname(),
-                'last_name'       => $shippingAddress->getLastname(),
-                'company'         => $shippingAddress->getCompany(),
-                'phone'           => $shippingAddress->getTelephone(),
-                'street_address1' => $shippingAddress->getStreetLine(1),
-                'street_address2' => $shippingAddress->getStreetLine(2),
-                'locality'        => $shippingAddress->getCity(),
-                'region'          => $shippingAddress->getRegion(),
-                'postal_code'     => $shippingAddress->getPostcode(),
-                'country_code'    => $shippingAddress->getCountryId(),
-            ];
+            if (! $clone->isVirtual()) {
+
+                // assign parent shipping method to clone
+                $address->setCollectShippingRates(true);
+                $address->setShippingMethod($quote->getShippingAddress()->getShippingMethod());
+
+                $this->totalsCollector->collectAddressTotals($clone, $address);
+                $address->save();
+
+                // Shipping address
+                $shipping_address = [
+                    'first_name' => $address->getFirstname(),
+                    'last_name' => $address->getLastname(),
+                    'company' => $address->getCompany(),
+                    'phone' => $address->getTelephone(),
+                    'street_address1' => $address->getStreetLine(1),
+                    'street_address2' => $address->getStreetLine(2),
+                    'locality' => $address->getCity(),
+                    'region' => $address->getRegion(),
+                    'postal_code' => $address->getPostcode(),
+                    'country_code' => $address->getCountryId(),
+                ];
 
-            $email = $shippingAddress->getEmail() ?: $email;
-            if ($email) {
-                $shipping_address['email'] = $email;
-            }
+                $email = $address->getEmail() ?: $email;
+                if ($email) {
+                    $shipping_address['email'] = $email;
+                }
 
-            foreach ($this->required_address_fields as $field) {
-                if (empty($shipping_address[$field])) {
-                    unset($shipping_address);
-                    break;
+                foreach ($this->required_address_fields as $field) {
+                    if (empty($shipping_address[$field])) {
+                        unset($shipping_address);
+                        break;
+                    }
                 }
-            }
 
-            if (@$shipping_address) {
-                $cost         = $shippingAddress->getShippingAmount();
-                $rounded_cost = $this->getRoundAmount($cost);
+                if (@$shipping_address) {
```

Comment:
looks like this `$shipping_address` is always defined when code executes here?

Reply:
it is conditionally unset a few lines above